10-JavaScript/fun-with-functions.md

```diff
@@ -1,34 +1,99 @@
+
+Either way, the syntax for functions is a lot different in JavaScript than in Ruby.
+
+## Functions Are Objects, So We Assign Variables to Them
+
+### Declare
+
+All functions in JavaScript are objects. For now, what that means to us is that we will make sure that variables hold onto these functions. You declare functions with the `let` or `const` keywords and a name.
 
-## Using Functions in JavaScript
-The syntax for functions is a lot different in JavaScript than in Ruby.
+### Assign
 
-### Defining Functions
-All functions in JavaScript are variables. Just like any other variable, you declare functions with the `let` or `const` keywords and a name, but you set the value to be a function instead of a number or a string.
+On the right-hand side of the assignment operator, you set the value to be a function, with the `function` keyword.
 
-How do you set the value to be a function? With the `function` keyword of course!
+Let's make a function named `bark`, whose responsibility is to print to the terminal the sound a dog makes.
+
+How do we do that? Let's declare a `const` variable named `bark`, and set its value to the function named `bark`. After we start the function with the words `function bark()`, we open up a block of code with curly braces, and put the inners of the function within the `{}`.
 
 ```javascript
 const bark = function bark() {
   console.log('Woof!');
 };
 ```
 
-**Note** we declared the function `bark` here with `const`.  We could have used `let` or `var`, but because we don't want `bark` to be reassigned, it makes sense to make it a constant.  We also added the function name to the right of the `function` keyword.  This is optional, but it can allow us to provide a [longer-more descriptive name for the stack trace](https://github.com/airbnb/javascript/issues/794).
+Open the `node` REPL and copy-and-paste in the following expressions. What do you get?
+
+1.
+    ```javascript
+    const bark = function bark() {
+      console.log('Woof!');
+    };
+    ```
+1. `bark;`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
 
-#### Calling a Function
-Unlike in Ruby, JavaScript requires you to type in parentheses if you want to invoke a function. You can access functions like you would any other variable:
+  1. `undefined`
+  1. `[Function: bark]`
+</details>
+
+### Calling a Function
+
+We've defined a function, so how do we call it?
+
+Unlike in Ruby, JavaScript requires you to type in parentheses if you want to invoke a function:
+
+```javascript
+bark();
+```
+
+### The Finer Details About Defining Functions
+
+When we defined the function, we gave it the variable keyword `const`. We could have used `let` or `var`, but in general, we don't need any variable named `bark` to be reassigned. Because we don't need `bark` to every be reassigned, it makes sense to make it a constant.
```

Comment:
I'd like to avoid mentioning `var` more than we have to.

```suggestion
When we defined the function, we gave it the variable keyword `const`. We could have used `let`, but in general, we don't need any variable named `bark` to be reassigned. Because we don't need `bark` to ever be reassigned, it makes sense to make it a constant.
```